### ISSUE TITLE
Strings: Code generation dirties other files

### ISSUE THREAD

**Issue by Robuske** [SwiftGen/SwiftGen#810]
Hello, while I was debugging a problem with _Sourcery_, [I found out that for some reason the `strings` subcommand dirties other files](https://github.com/krzysztofzablocki/Sourcery/issues/879#issuecomment-747826770), dirties in the sense that it makes _Sourcery_ think that the file was modified and needs to be parsed again.

We have a _Makefile_ for code generation, it always runs _SwiftGen_ and _Sourcery_. Thanks to some tests I realized that every time `strings` generates any code it makes _Sourcery_ think it needs to parse all files again.
Weirdly enough, the `assets` subcommand does not trigger this, I even tested if it was `rm` or `make` itself causing the problem, but no, only happens when `strings` **generates code again** (if it detects that the content is unchanged it does not trigger too).

```shell
$ swift --version
Apple Swift version 5.3.2 (swiftlang-1200.0.45 clang-1200.0.32.28)
Target: x86_64-apple-darwin20.1.0

$ xcode-select --version
xcode-select version 2384. # This is Xcode 12.3 (12C33)

$ swiftgen --version
SwiftGen v6.4.0 (Stencil v0.13.1, StencilSwiftKit v2.7.2, SwiftGenKit v6.4.0)

$ sourcery --version
1.0.2
```

Booth installed trough _Homebrew_.

---

Our _Makefile_:

```makefile
SWIFTGEN_DIR=./Tools/SwiftGen
GENERATED_CODE_DIR=./Modules/Sources/CoreUI-Resources/Generated
GENERATED_SNAPSHOTS_DIR=./Modules/Tests/PreviewsSnapshots-Tests

all: generate_assets_code generate_strings_code generate_snapshots_code

clean: clean_assets_code clean_strings_code clean_snapshots_code

# Assets

generate_assets_code:
	swiftgen config run --config $(SWIFTGEN_DIR)/assets.yml

clean_assets_code:
	rm -f $(GENERATED_CODE_DIR)/Asset.generated.swift

# Strings

generate_strings_code:
	swiftgen config run --config $(SWIFTGEN_DIR)/strings.yml

clean_strings_code:
	rm -f $(GENERATED_CODE_DIR)/L10n.generated.swift

# Snapshots

generate_snapshots_code: 
	sourcery

clean_snapshots_code:
	rm -f $(GENERATED_SNAPSHOTS_DIR)/*.generated.swift
```

---

_strings.yml_:

```yml
input_dir: ../../App/Resources/
output_dir: ../../Modules/Sources/CoreUI-Resources/Generated/

strings:
  inputs: Localizable.strings
  outputs:
    templateName: structured-swift5
    output: L10n.generated.swift
    params:
      publicAccess: true
      bundle: Bundle.main
```

---

_assets.yml_:

```yml
input_dir: ../../App/Resources/
output_dir: ../../Modules/Sources/CoreUI-Resources/Generated/

xcassets:
  inputs: ModulesAssets.xcassets
  outputs:
    templateName: swift5
    output: Asset.generated.swift
    params:
      forceProvidesNamespaces: true
      publicAccess: true
      bundle: Bundle.main
```

---

Should not be relevant, but _.sourcery.yml_:

```yml
sources:
  - Modules/Sources
templates:
  - Tools/Sourcery
output: Modules/Tests/
prune: true
```

**Comment by djbe** [comment]
I'm not sure what Sourcery uses to check if files have changed, but I assume it's either the checksum of a file, or the timestamp of a file. Have you checked if, when the strings command generates code, the timestamp/checksum of the assets file changes?

I'm not sure this "can" even be a SwiftGen issue, because your strings config makes no mention of the assets files/output file, so SwiftGen wouldn't even know how to modify those files.

As a side note: having multiple configs, one for each asset/parser is less efficient than having 1 config, as SwiftGen won't be able to execute parsers in parallel.

**Comment by Robuske** [comment]
> I'm not sure this "can" even be a SwiftGen issue, because your strings config makes no mention of the assets files/output file, so SwiftGen wouldn't even know how to modify those files.

I found it odd too, but it was the only way I found that could reproducing the issue, and it reproduces consistently.

> Have you checked if, when the strings command generates code, the timestamp/checksum of the assets file changes?

How can I check that with more precision? I looked at the file modified date in its _Get Info_ menu, but it hadn't changed.

> As a side note: having multiple configs, one for each asset/parser is less efficient than having 1 config, as SwiftGen won't be able to execute parsers in parallel.

I thought that could be the case, but it executes so fast (and only after checking out a branch) that doesn't really matter, so we left like that just to have a clearer distinction between them.

**Comment by Robuske** [comment]
I just wanted to report that the problem seems to have ben fixed, now booth _SwiftGen_ and _Sourcery_ execute in a few seconds.

I am not sure exactly when it was fixed, I am still using _SwiftGen_ 6.4.0, but _Sourcery_ is now on 1.4.1, looking through their releases I think it may have been the [1.3.0 release](https://github.com/krzysztofzablocki/Sourcery/releases/tag/1.3.0), where they changed from _SourceKit_ to _SwiftSyntax_.
